src/pool/srv_target.c

```diff
@@ -637,6 +642,37 @@ pool_tgt_query(struct ds_pool *pool, struct daos_pool_space *ps)
 	coll_args.ca_aggregator		= &agg_arg;
 	coll_args.ca_func_args		= &coll_args.ca_stream_args;
 
+	if (pool->sp_map) {
+		d_rank_t	myrank;
+		unsigned int	tgt_cnt;
+
+		/* Check if we need excluded the failure targets */
+		rc = crt_group_rank(pool->sp_group, &myrank);
```

Comment:
seems better to use crt_group_rank(NULL, &myrank);
if the pool created on subset of primary group (for example on rank 3,4,5,6), then the rank in the pool group is different with rank in primary group, but rank in pool map should be for primary group.

Reply:
ah, you are right. Thanks.